test/model_simulation/simulate_SDEs.jl

```diff
@@ -119,31 +119,122 @@ let
-                                 noise_scaling = (@variables η3)[1]), ImplicitEM())
-        @test 2 * std(first.(sol001.u)[100:end]) < std(first.(sol01.u)[100:end])
-        @test 2 * std(last.(sol001.u)[100:end]) < std(last.(sol01.u)[100:end])
-        @test 2 * std(first.(sol01.u)[100:end]) < std(first.(sol1.u)[100:end])
-        @test 2 * std(last.(sol01.u)[100:end]) < std(last.(sol1.u)[100:end])
-        @test 2 * std(first.(sol1.u)[100:end]) < std(first.(sol10.u)[100:end])
-        @test 2 * std(last.(sol1.u)[100:end]) < std(last.(sol10.u)[100:end])
+    η_stored = :η
+    @variables t
+    @species X1(t) X2(t)
+    p_syms = @parameters $(η_stored) k1 k2
+
+    r1 = Reaction(k1,[X1],[X2],[1],[1]; metadata = [:noise_scaling => η_stored])
+    r2 = Reaction(k2,[X2],[X1],[1],[1]; metadata = [:noise_scaling => η_stored])
+    @named noise_scaling_network = ReactionSystem([r1, r2], t, [X1, X2], [k1, k2, p_syms[1]])
+
+    u0 = [:X1 => 1100.0, :X2 => 3900.0]
+    p = [:k1 => 2.0, :k2 => 0.5, :η=>0.0]
+    @test_broken SDEProblem(noise_scaling_network, u0, (0.0, 1000.0), p).ps[:η] == 0.0 # Broken due to SII/MTK stuff.
+end
+
+# Complicated test with many combinations of options.
+# Tests the noise_scaling_parameters getter.
+let
+    noise_scaling_network = @reaction_network begin 
+        @parameters k1 par1 [description="Parameter par1"] par2 η1 η2=0.0 [description="Parameter η2"] η3=1.0 η4
+        (p, d), 0 ↔ X1, ([noise_scaling=η1],[noise_scaling=η2]) 
+        (k1, k2), X1 ↔ X2, ([noise_scaling=η3],[noise_scaling=η4])  
     end
+    u0 = [:X1 => 500.0, :X2 => 500.0]
+    p = [:p => 20.0, :d => 0.1, :η1 => 0.0, :η3 => 0.0, :η4 => 0.0, :k1 => 2.0, :k2 => 2.0, :par1 => 1000.0, :par2 => 1000.0]
+    
+    @test getdescription(parameters(noise_scaling_network)[2]) == "Parameter par1"
+    @test getdescription(parameters(noise_scaling_network)[5]) == "Parameter η2"
+    
+    sprob = SDEProblem(noise_scaling_network, u0, (0.0, 1000.0), p)
+    @test sprob.ps[:η1] == sprob.ps[:η2] == sprob.ps[:η3] == sprob.ps[:η4] == 0.0
 end
 
+# Tests default_noise_scaling_option.
+# Tests using sys. indexing for setting noise scaling parameter values.
+# tests for default value used in noise scaling parameter.
+let
+    noise_scaling_network = @reaction_network begin
+        @parameters η1 η2=0.1
+        @default_noise_scaling η1
+        (p,d), 0 <--> X1, ([noise_scaling=η2],[noise_scaling=η2])
+        (p,d), 0 <--> X2, ([description="Y"],[description="Y"])
+        (p,d), 0 <--> X3
+    end
+    noise_scaling_network = complete(noise_scaling_network)
+    
+    u0 = [:X1 => 1000.0, :X2 => 1000.0, :X3 => 1000.0]
+    ps = [noise_scaling_network.p => 1000.0, noise_scaling_network.d => 1.0, noise_scaling_network.η1 => 1.0]
+    sol = solve(SDEProblem(noise_scaling_network, u0, (0.0, 1000.0), ps), ImplicitEM(); saveat=1.0)
+    @test var(sol[:X1]) < var(sol[:X2]) 
+    @test var(sol[:X1]) < var(sol[:X3]) 
```

Comment:
Would it not be better to just write a function to extract the noise coefficient from each equation in the generated `SDESystem` and then check it is correct for each term? That seems more robust than the `sol` test here (which I don't really see much value to, and seems like it could randomly fail on us).

Reply:
This has the same issue with the test potentially failing doesn't it?